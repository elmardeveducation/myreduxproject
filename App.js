import React, { Component } from "react";
import { StyleSheet,} from "react-native";
import CounterApp from './src/CounterApp'

/**
 * Store - holds our state - THERE IS ONLY ONE STATE 
 * Action - State can be modified using actions - SIMPLE OBJECTS 
 * Dispatcher - Action needs to be sent by someone - known as dispatching an action
 * Reducer - receives the action and modifies the state to give us a new state 
 *  - pure functions 
 *  - only mandatory argument is the 'type' 
 * Subscriber - listens for state change to update the ui  
 */

class App extends Component {

  render() {
    return (
      <CounterApp />

    );
  }
}

export default App

// export default App;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  }
});