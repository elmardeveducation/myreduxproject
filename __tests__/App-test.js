/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { exportAllDeclaration } from '@babel/types';
import { JestEnvironment } from '@jest/environment';
jest.mock('../src/CounterApp', ()=>()=>{return (null)})

describe('test my all APP component', ()=>{

//snapshot my component App
  it('renders correctly', () => {
  let mysnapshotforApp=renderer.create(<App />).toJSON();
  console.log(mysnapshotforApp)
  expect (mysnapshotforApp).toMatchSnapshot()
});


})