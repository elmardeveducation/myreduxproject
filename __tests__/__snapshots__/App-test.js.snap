// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test my all APP component renders correctly 1`] = `null`;
